frontend/src/components/BCDataGrid/BCColumnSetFilter.jsx

```diff
@@ -0,0 +1,137 @@
+import {
+  forwardRef,
+  useRef,
+  useImperativeHandle,
+  useState,
+  useEffect
+} from 'react'
+import { Autocomplete, TextField, Box, Checkbox } from '@mui/material'
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank'
+import CheckBoxIcon from '@mui/icons-material/CheckBox'
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />
+const checkedIcon = <CheckBoxIcon fontSize="small" />
+
+const BCColumnSetFilter = forwardRef((props, ref) => {
+  const { columnWidth } = props.column.colDef
+  const [options, setOptions] = useState([])
+  const { data: optionsData, isLoading: optionsIsLoading } = props.apiQuery()
+
+  const [currentValue, setCurrentValue] = useState(null)
+  const inputRef = useRef(null)
+
+  // expose AG Grid Filter Lifecycle callbacks
+  useImperativeHandle(ref, () => {
+    return {
+      onParentModelChanged(parentModel) {
+        // When the filter is empty we will receive a null value here
+        if (!parentModel) {
+          inputRef.current.value = ''
```

Comment:
do we need to set the ref value directly? It's a bit of a anti-pattern in react, ideally we handle this with state. Let me know what you think.

Reply:
Added for the testing purpose but now removed it.